src/ts/uimanager.ts

```diff
@@ -1001,6 +1001,97 @@ export namespace UIManager.Factory {
   }
 }
 
+// Separate the demos from the other factories.
+// The others are more like different 'styles' than some variants with different features
+export namespace UIManager.Demo.Factory {
```

Comment:
Please put this into a separate file. I understand why you put it here, but it's actually not desired to have everything within this file. The `Factory` and other classes will be extracted into separate files with UI v3. They have just been remaining here for the time being (UI v2) to minimize merge conflicts for customers with custom branches when they update the UI framework.

Reply:
Is it possible to easy split it?

The root namespace and the class have the same name -> `UIManager`. 
When you split it, you have to import the `UIManager` class but this will results in an error `Individual declarations in merged declaration 'UIManager' must be all exported or all local.` because of the naming conflict.

Also the export is needed (comment above).

Or am i missing something here.

Reply:
Just move it to a file called `demofactory.ts` and `export namespace DemoFactory` in there (an add the export into `main.ts`. There's (hacky) ways to achieve what you attempted, but it does not have to be under the `UIManager` namespace so let's take the clean approach.